docs/en/query_language/functions/array_functions.md

```diff
@@ -814,6 +814,27 @@ SELECT arrayReduce('uniqUpTo(3)', [1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
 └─────────────────────────────────────────────────────────────┘
 ```
 
+## arrayReduceInRanges(agg_func, indices, lengths, arr1, ...) {#array_functions-arrayreduceinranges}
+
+Applies an aggregate function to array elements in given ranges and returns an array containing the result corresponding to each range. The function will return the same result as multiple `arrayReduce(agg_func, arraySlice(arr1, index, length), ...)`.
+
+Example:
+
+```sql
+SELECT arrayReduceInRanges(
+    'sum',
+    [1, 2, 3, 4],
+    [5, 4, 4, 4],
+    [1000000, 200000, 30000, 4000, 500, 60, 7]
+) AS res
+```
+
+```text
+┌─res─────────────────────────┐
+│ [1234500,234500,34560,4567] │
+└─────────────────────────────┘
+```
+
```

Comment:
It is better to explicitly describe parameters.
By the way there is a function doc template:
https://github.com/ClickHouse/ClickHouse/blob/master/docs/dscr-templates/template-function.md